test/test_type_promotion.py

```diff
@@ -227,6 +227,10 @@ def test_result_type(self):
         self.assertEqual(torch.result_type(torch.tensor([1., 1.], dtype=torch.float), 1.), torch.float)
         self.assertEqual(torch.result_type(torch.tensor(1., dtype=torch.float), torch.tensor(1, dtype=torch.double)), torch.double)
 
+    def test_can_cast(self):
+        self.assertTrue(torch.can_cast(torch.double, torch.float))
+        self.assertFalse(torch.can_cast(torch.float, torch.int))
```

Comment:
I don't, do you think it'd be helpful? it's called/used pretty heavily in the actual arithmetic operators and in tensoriterator, which does have a lot of python test coverage.

Reply:
Hmm ideally we'd unit test heavily-used helper functions like `canCast` during development instead of testing them indirectly by testing arithmetic operators so that we can be sure that we're building on top of working code. But since we have all of those tests for arithmetic operators and type promotion this should be fine.